src/builtins/jobs.cpp

```diff
@@ -42,8 +42,11 @@ static double cpu_use(const job_t *j) {
 /// Print information about the specified job.
 static void builtin_jobs_print(const job_t *j, int mode, int header, io_streams_t &streams) {
     int pgid = INVALID_PID;
-    if (auto job_pgid = j->get_pgid()) {
-        pgid = *job_pgid;
+    {
+        auto job_pgid = j->get_pgid();
+        if (job_pgid.has_value()) {
```

Comment:
We could probably upgrade to C++17 soon and use `if` init-statements

```cpp
if (auto job_pgid = j->get_pgid(); job_pgid.has_value()) {
```

Reply:
The cutoff point on C++17 support seems to be gcc 7 (https://gcc.gnu.org/projects/cxx-status.html#cxx17 - with one teensy exception in gcc 8).

According to https://repology.org/project/gcc/versions, that's CentOS/RHEL 8, Version 7 ships with gcc 4.8.5 and is supported until 2024.

So we would be asking people there to upgrade their compiler - which includes us, since we build for CentOS on OBS.

Reply:
I'm leery of upgrading to a newer spec because I was around for the introduction of C++11 and saw how long it took for compilers to actually get their act together and actually support what they claimed to. I'm sure there are quality of life benefits in C++14 and C++17 but I'm honestly continuing to use the sane subset of C++11 we already do. C++11 vs C++98 was huge and a fundamentally seismic shift in making the standard library usable to write cross-platform software, but I don't think we can say the same about the newer variants (for now). Also, it's a stupid reason but I love that I can dig out old VMs running ancient OSes and get fish to compile and I would genuinely miss being able to do that.

(Though I have to say C++20 concepts would have made disabling bool conversion for bool-convertible types here so much easier than the SIFNAE hell I ended up writing. Apparently you can just write your member function and after the prototype add `requires (!std::is_convertible<T>)` which is nice, but not really world changing and not worth breaking compatibility over.)

Reply:
I built new compilers for old images on OBS and I did not enjoy it!